subsys/net/lib/sockets/sockets.c

```diff
@@ -1337,13 +1385,19 @@ static inline ssize_t zsock_recv_dgram(struct net_context *ctx,
 	if (src_addr && addrlen) {
 		if (IS_ENABLED(CONFIG_NET_OFFLOAD) &&
 		    net_if_is_ip_offloaded(net_context_get_iface(ctx))) {
-			/*
-			 * Packets from offloaded IP stack do not have IP
-			 * headers, so src address cannot be figured out at this
-			 * point. The best we can do is returning remote address
-			 * if that was set using connect() call.
-			 */
-			if (ctx->flags & NET_CONTEXT_REMOTE_ADDR_SET) {
+			if (!net_pkt_addr_is_unspecified(pkt)) {
+				int rv;
```

Comment:
Please use `ret` instead of rv (to be consistent with the rest of the file).

Reply:
renamed